### DISCUSSION TITLE
Is check per customer usage limit for coupons executed twice?

### DISCUSSION THREAD

**Discussion by seansan** [OpenMage/magento-lts#2104] [Questions & Answers]
Is this check 
// check per rule usage limit on line 205 
the same as -- only if coupon code used
// check per customer usage limit on line 185

and thus executed twice?

https://github.com/OpenMage/magento-lts/blob/77a837aba9166502378d3bb64920ccf5e862b95e/app/code/core/Mage/SalesRule/Model/Validator.php#L205

**Comment by mattdavenport** [comment]
These two checks are looking at different tables. The first is checking `salesrule_coupon_usage`, and the next one is `salesrule_customer`. I believe this is checking the hierarchy of:

1. Usage per customer (coupon)
2. Usage per rule (per customer)
